src/software/geom/util.h

```diff
@@ -374,6 +374,19 @@ Point reflect(const Point &v, const Point &n);
 std::pair<std::optional<Point>, std::optional<Point>> raySegmentIntersection(
     const Ray &ray, const Segment &segment);
 
+/**
+ * Calculates the intersection of a Ray and Rectangle
+ *
+ * @param ray The ray
+ * @param rectangle The rectangle
+ * @return Returns {std::nullopt, std::nullopt} if no intersections exist.
+ * Returns {Point, std::nullopt} if a single intersection exists.
+ * Returns {Point, Point} if the ray overlaps a segment of the rectangle are overlapping,
+ * where the points define the line segment of overlap.
```

Comment:
1. Check wording here.
2. This should ideally return 2 intersections if the ray crosses 2 sides of the rectangle and it should expand to cover all convex shapes. Can we file an issue for that?

Reply:
wording fixed.

I believe this does return 2 intersection points if 2 sides of the rectangle are intersected.

I'm not going to make a new ticket yet until Sam and Jason have started some high-level geometry tickets and made a plan (so we don't just have a random ticket). This should be addressed in the geometry cleanup regardless.